### PR TITLE
Remove classnames dependency, use es6 modules

### PR DESCRIPTION
**Pull request by swese44**
## Pull request checklist

* [x] Component `README.md` file is up-to-date.
* [x] Component is unit tested.

### CODE REVIEW

File: src/util/classNames.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+export type ClassName = string | undefined;
+
+export const join = (classNames: ClassName[]): string => {
+  if (!classNames.length) {
```
Comment (unindented) [inline src/util/classNames.ts]: I'm not a fan of type coercions, I'd have explicitly compared with `=== 0`.

File: tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
-    "target": "es5",
-    "module": "commonjs",
+    "target": "es6",
```
Comment (unindented) [inline tsconfig.json]: Why also change the target? If you change the target, then consumers will have to transpile. If you leave the target as `es5` and change the module type to `es6`, webpack will be able to do the right thing.

**Comment by swese44** [comment]
udpated

File: src/components/FakeLink/FakeLink.tsx [hunk 3]
```diff
@@ -19,6 +19,6 @@ export default class FakeLink extends React.Component<FakeLinkProps, {}> {
   render() {
     const { className, children } = this.props;
 
-    return <span className={classNames('y-fakeLink', className)}>{children}</span>;
+    return <span className={join(['y-fakeLink', className])}>{children}</span>;
```
Comment (richardtagger) [inline src/components/FakeLink/FakeLink.tsx]: why not just use Array.prototype's join?

Comment (richardtagger) [reply src/components/FakeLink/FakeLink.tsx]: ah I see, it is underneath. k

Comment (swese44) [reply src/components/FakeLink/FakeLink.tsx]: this removes empty/undefined items so we don't have to check for the existence of `className` in each component

**Pull request merged**